### ISSUE TITLE
Retrolambda can not be run under JDK 9

### ISSUE THREAD

**Issue by teras** [orfjackal/retrolambda#137]
If we run retrolambda under Java 9, then this error appears:

```00:00 ERROR: Failed to run Retrolambda
java.lang.IncompatibleClassChangeError: Method java.util.stream.Stream.of([Ljava/lang/Object;)Ljava/util/stream/Stream; must be InterfaceMethodref constant
	at net.orfjackal.retrolambda.SystemPropertiesConfig.parsePathList(SystemPropertiesConfig.java:158)
	at net.orfjackal.retrolambda.SystemPropertiesConfig.getClasspath(SystemPropertiesConfig.java:148)
	at net.orfjackal.retrolambda.Retrolambda.run(Retrolambda.java:25)
	at net.orfjackal.retrolambda.Main.main(Main.java:28)
```

Compared with issue https://github.com/orfjackal/retrolambda/issues/119 indeed at `net.orfjackal.retrolambda.SystemPropertiesConfig.parsePathList(SystemPropertiesConfig.java:158)` there is a call to Stream.of which, since retrolamda jar is downgraded, it falls to the same problem. 

Apart of the obvious answer ("run retrolambda under Java 8), is there any real solution to this problem?

**Comment by orfjackal** [comment]
This problem originates from Retrolambda's build processing itself with Retrolambda so that the Maven plugin, which is Java 6 compatible, can be compiled to call Retrolambda's APIs directly. Some options for fixing this:

- Move those classes that the Maven plugin uses (SystemPropertiesConfig) into another module which is compiled using Java 6 (calling the main method probably needs reflection), and do not run Retrolambda on itself.
- Avoid using the Stream API inside Retrolambda and instead use the streamsupport library.
- Downgrade Retrolambda's code to pre-Java 8.

I'll try out the first option and see how it goes...

**Comment by teras** [comment]
I think the problem could be easily solved by replacing
```
        return Stream.of(paths.split(File.pathSeparator))
```
with 
```
        return Arrays.asList(paths.split(File.pathSeparator)).stream()
```
at `retrolambda/retrolambda/src/main/java/net/orfjackal/retrolambda/SystemPropertiesConfig.java` line 158

**Comment by orfjackal** [comment]
This has been fixed in Retrolambda 2.5.2

**Comment by Nighty42** [comment]
Had this issue in Gradle 4.4.1 - currently the most recent version. Gradle 4.4.1 uses Retrolambda 2.5.1 so I hope they add the fixed version soon. I reported this issue here: [Gradle - Issue 3921](https://github.com/gradle/gradle/issues/3921)
